model/task/task.go

```diff
@@ -775,6 +778,39 @@ func (t *Task) AllDependenciesSatisfied(cache map[string]Task) (bool, error) {
 	return true, nil
 }
 
+// MarkDependenciesFinished updates all direct dependencies on this task to
+// cache whether or not this task has finished running.
+func (t *Task) MarkDependenciesFinished(finished bool) error {
+	if t.DisplayOnly {
+		// This update can be skipped for display tasks since tasks are not
+		// allowed to have dependencies on display tasks.
+		return nil
+	}
+
+	env := evergreen.GetEnvironment()
+	ctx, cancel := env.Context()
+	defer cancel()
+
+	_, err := env.DB().Collection(Collection).UpdateMany(ctx,
+		bson.M{
+			DependsOnKey: bson.M{"$elemMatch": bson.M{
+				DependencyTaskIdKey: t.Id,
+			}},
+		},
+		bson.M{
+			"$set": bson.M{bsonutil.GetDottedKeyName(DependsOnKey, "$[elem]", DependencyFinishedKey): finished},
+		},
+		options.Update().SetArrayFilters(options.ArrayFilters{Filters: []interface{}{
+			bson.M{bsonutil.GetDottedKeyName("elem", DependencyTaskIdKey): t.Id},
+		}}),
+	)
+	if err != nil {
+		return errors.Wrap(err, "marking finished dependencies")
```

Comment:
Can we change this error to `error setting finished dependencies to '%s', finished`?

Reply:
I intentionally don't write my error messages starting with "failed to do xyz" or "error doing xyz" or "cannot do xyz" because they tend to produce excessively verbose error messages. Instead, I write them in the style of the Go standard library (which is the golden standard for Go code), which, when wrapping errors with more context, says what operation was being done but doesn't produce ugly wrapped errors that look like "error doing abc: failed to do def: could not do xyz: ". My experience with this is that it's much easier to read error messages in logs without this noise.

This error message style isn't consistent across Evergreen teams, so I just use the preference of PLT, which is no noisy error messages.

Reply:
That makes sense. Do you think it would be worth creating a tech debt ticket to modify the error messages across our codebase to use this style?

Reply:
Sure, I made [EVG-16501](https://jira.mongodb.org/browse/EVG-16501) to clean it up in all the Evergreen-owned repos and [EVG-16504](https://jira.mongodb.org/browse/EVG-16504) to clean it up in Evergreen itself.